kale/utils/distance.py

```diff
@@ -0,0 +1,44 @@
+# =============================================================================
+# Author: Sina Tabakhi, sina.tabakhi@gmail.com
+# =============================================================================
+
+"""
+Provide distance (i.e., similarity) calculation methods using various distance metrics.
+"""
+
+from enum import Enum
+
+import torch
+
+
+class DistanceMetric(Enum):
+    COSINE = "COSINE"
+
+
+def calculate_distance(
+    x1: torch.Tensor, x2: torch.Tensor = None, eps: float = 1e-8, metric: DistanceMetric = DistanceMetric.COSINE
+) -> torch.Tensor:
+    r"""Returns similarity between :math:`x_1` and :math:`x_2`, computed along `dim`=1. This method calculates the
+    similarity between each pair of data points in two input matrices.
+
+    Note that this implementation differs from the existing implementations in
```

Comment:
@XianyuanLiu Can you please take a look at this and let me know your opinion? Thanks.

Reply:
Shall we use existing functions from other libraries? e.g., cosine similarity from [Torch.nn](https://pytorch.org/docs/stable/generated/torch.nn.functional.cosine_similarity.html), [Scikit Learn](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html), and [TorchMetrics](https://torchmetrics.readthedocs.io/en/stable/regression/cosine_similarity.html). Do they achieve the functions we need？

Reply:
@SinaTabakhi Please have a discussion with Xianyuan

Reply:
According to @shuo-zhou, Scikit-Learn has a general function that provides many functions of [pairwise distance](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise_distances.html#sklearn.metrics.pairwise_distances), while Torch does not. We may create one in the future.